services/src/main/java/org/keycloak/protocol/oidc/endpoints/LogoutEndpoint.java

```diff
@@ -166,13 +168,25 @@ public Response logout(@QueryParam(OIDCLoginProtocol.REDIRECT_URI_PARAM) String
             return ErrorPage.error(session, null, Response.Status.BAD_REQUEST, Messages.INVALID_PARAMETER, OIDCLoginProtocol.REDIRECT_URI_PARAM);
         }
 
-        if (postLogoutRedirectUri != null && encodedIdToken == null) {
+        if (postLogoutRedirectUri != null && encodedIdToken == null && clientId == null) {
             event.event(EventType.LOGOUT);
             event.error(Errors.INVALID_REQUEST);
-            logger.warnf("Parameter 'id_token_hint' is required when 'post_logout_redirect_uri' is used.");
+            logger.warnf("Either the parameter 'client_id' or the parameter 'id_token_hint' is required when 'post_logout_redirect_uri' is used.");
             return ErrorPage.error(session, null, Response.Status.BAD_REQUEST, Messages.MISSING_PARAMETER, OIDCLoginProtocol.ID_TOKEN_HINT);
         }
 
+        boolean confirmationNeeded = true;
+        boolean forcedConfirmation = false;
+        ClientModel client = clientId == null ? null : realm.getClientByClientId(clientId);
+        if (clientId != null) {
+            client = realm.getClientByClientId(clientId);
+            // Forced confirmation if client referenced by client_id was not found
+            if (client == null) {
+                logger.warnf("Client '%s' not found.", clientId);
+                forcedConfirmation = true;
+            }
+        }
```

Comment:
Perhaps, it'd be better? It's not necessary to fetch the client again, right?

```suggestion
        if (clientId != null && client == null) {
            logger.warnf("Client '%s' not found.", clientId);
            forcedConfirmation = true;
        }
```

Reply:
@mabartos Good catch! During the refactoring, I did the unneccessary duplicated retrieval of the client. Agree that this can be skipped.